lib.include.sh

```diff
@@ -203,12 +204,25 @@ function has_conda_global_env {
 
 function create_conda_env {
     print "Creating Conda environment in \"${OT_CONDA_ENV}\"..."
+
     # IMPORTANT: The ".*" suffix tells Conda to install the latest bugfix/patch
     # release of the desired Python version. For example, if we specify "3.12.*",
     # then it will pick the latest patch release, such as "3.12.5". It also works
     # correctly if we specify an EXACT patch release ourselves, such as "3.10.14.*",
     # or if we only specify a major version, such as "3.*" (gets the latest release).
-    run_conda create -y --prefix "${OT_CONDA_ENV}" "python==${OT_CONDA_USE_PYTHON_VERSION}.*"
+    declare -a install_args=()
+    install_args+=("python==${OT_CONDA_USE_PYTHON_VERSION}.*")
+
+    # IMPORTANT: We MUST use "conda-forge" and EXPLICITLY switch to the version
+    # of Tk that has libXft support on Linux, otherwise the GUI will have broken
+    # fonts, inability to render Unicode, and no antialiasing! Doesn't affect Macs.
+    # SEE: https://github.com/conda-forge/tk-feedstock/pull/40#issuecomment-2381409555
+    # SEE: https://anaconda.org/conda-forge/tk/files (only Linux has "xft" variant).
+    if [[ "${OT_HOST_OS}" == "Linux" ]]; then
+        install_args+=("tk[build=xft_*]")
+    fi
+
+    run_conda create -y --prefix "${OT_CONDA_ENV}" --channel "conda-forge" "${install_args[@]}"
```

Comment:
Yeah I am very aware that `defaults` is added implicitly, and that it can be disabled with `--override-channels` too (this ensures we only use conda-forge).

But the reason I didn't do that is because Conda will search `--channel` first, and then fall back to defaults if channel is MISSING the thing we need. So we _need_ to keep defaults as fallback.

With the current PR, we already get all packages via conda-forge.

However, I noticed one improvement we can make, and this is what Conda-Forge themselves recommend:

```
  --strict-channel-priority
                        Packages in lower priority channels are not considered
                        if a package with the same name appears in a higher
                        priority channel.
```

It ensures that if a high-priority channel (conda-forge) has a package such as `tk 8.0` and requirements specify `tk>=9` and `tk 9.0` exists in defaults, then it will throw an error instead of falling back to the other channels. I will do that change.

Reply:
If anyone's curious, here's how Conda-Forge recommends installing, with strict priority but defaults stil lenabled:

https://conda-forge.org/docs/user/introduction/#how-can-i-install-packages-from-conda-forge

Interestingly, they also recommend totally avoiding the Anaconda distribution, because THAT can cause problems apparently. Good thing I already made our recommendation Miniconda instead, since it's far superior (smaller download).

> We also recommend the use of [miniforge](https://github.com/conda-forge/miniforge) instead of the Anaconda Distribution. Using environments make it easier to debug problems with packages and ensure the stability of your base environment. Avoiding the Anaconda Distribution reduces chances of unsolvable/conflicting installations, and it is also a smaller download.

Reply:
Let's add `--strict-channel-priority` then?

Reply:
Yep, pushed that change. It's a good idea to have that protection.